File: module_gift/src/main/java/com/miaomi/fenbei/gift/adapter/UserListAdapter.java
package com.miaomi.fenbei.gift.adapter;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.miaomi.fenbei.base.util.ImgUtil;
import com.miaomi.fenbei.gift.R;
import com.miaomi.fenbei.base.bean.GiftInfoBean;

import java.util.ArrayList;
import java.util.List;

public class UserListAdapter extends RecyclerView.Adapter<UserListAdapter.ViewHodler> {
    private Context context;
    private List<GiftInfoBean.ListBean> mList = new ArrayList<>();

    public UserListAdapter(Context context) {
        this.context = context;
    }

    public void setData(List<GiftInfoBean.ListBean> list){
        mList.clear();
        mList.addAll(list);
        notifyDataSetChanged();
    }
    public void setAllSeleted(){
        for (GiftInfoBean.ListBean bean :mList){
            bean.setSelected(true);
        }
        notifyDataSetChanged();
    }
    public void setAllUnSeleted(){
        for (GiftInfoBean.ListBean bean :mList){
            bean.setSelected(false);
        }
        notifyDataSetChanged();
    }

    public boolean isAllSeleted(){
        for (GiftInfoBean.ListBean bean :mList){
            if (!bean.isSelected()){
                return false;
            }
        }
        return true;
    }

    @NonNull
    @Override
    public ViewHodler onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHodler(LayoutInflater.from(context).inflate(R.layout.gift_item_user, parent, false));
    }

    private List<GiftInfoBean.ListBean> getSeletedList(){
        List<GiftInfoBean.ListBean> seletedList = new ArrayList<>();
        for (GiftInfoBean.ListBean bean :mList){
            if (bean.isSelected()){
                seletedList.add(bean);
            }
        }
        return seletedList;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHodler holder, final int position) {
        if (mList.get(position).getType() == 0){
            holder.seletedTv.setText("房主");
        }else if (mList.get(position).getType() == -1){
            holder.seletedTv.setText(mList.get(position).getNickname());
        }else{
            holder.seletedTv.setText(mList.get(position).getType()+"麦");
        }
        holder.seletedTv.setSelected(mList.get(position).isSelected());
        holder.avterIv.setSelected(mList.get(position).isSelected());
        ImgUtil.INSTANCE.loadCircleImg(context,mList.get(position).getFace(),holder.avterIv,R.drawable.common_avter_placeholder);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mList.get(position).isSelected()){
                    mList.get(position).setSelected(false);
                }else{
                    mList.get(position).setSelected(true);
                }
                notifyDataSetChanged();
                if (onAllSeletedListener != null){
                    onAllSeletedListener.onSeletedChange(getSeletedList());
                }
            }
        });
    }
    private OnSeletedChangeListener onAllSeletedListener;

    public void setOnSeletedChangeListener(OnSeletedChangeListener onAllSeletedListener) {
        this.onAllSeletedListener = onAllSeletedListener;
    }

    public interface OnSeletedChangeListener {
        void onSeletedChange(List<GiftInfoBean.ListBean> seletedList);
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }

    static class ViewHodler extends RecyclerView.ViewHolder{
        ImageView avterIv;
        TextView seletedTv;
        ViewHodler(View itemView) {
            super(itemView);
            avterIv = itemView.findViewById(R.id.iv_avter);
            seletedTv = itemView.findViewById(R.id.tv_selected);
        }
    }
}

File: module_room/src/main/java/com/miaomi/fenbei/room/ui/adapter/UserListAdapter.kt
package com.miaomi.fenbei.room.ui.adapter


import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.miaomi.fenbei.room.R
import com.miaomi.fenbei.room.ui.callback.WordClickListener
import com.miaomi.fenbei.base.bean.UserInfo
import com.miaomi.fenbei.base.util.ImgUtil
import kotlinx.android.synthetic.main.room_dialog_user_card.view.*
import kotlinx.android.synthetic.main.room_item_user_list.view.*
import kotlinx.android.synthetic.main.room_item_user_list.view.user_icon
import kotlinx.android.synthetic.main.room_item_user_list.view.user_id
import kotlinx.android.synthetic.main.room_item_user_list.view.user_nick
import kotlinx.android.synthetic.main.room_item_user_list.view.user_sex
import kotlinx.android.synthetic.main.room_item_user_list.view.wealth_level_iv


class UserListAdapter(list: ArrayList<UserInfo>, listener: WordClickListener, context: Context) : RecyclerView.Adapter<UserListAdapter.ViewHolder>() {
    private var mList: ArrayList<UserInfo> = list
    private var mListener: WordClickListener = listener
    private var mContext: Context = context


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.room_item_user_list, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(mList[position])
    }

    override fun getItemCount(): Int = mList.size

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(bean: UserInfo) {
            itemView.user_id.visibility = View.VISIBLE
            itemView.user_sex.visibility = View.VISIBLE
            itemView.wealth_level_iv.setWealthLevel(bean.wealth_level.grade)
            if (bean.good_number_state == 1){
                itemView.user_id.show(bean.good_number.toString(),true)
            }else{
                itemView.user_id.show(bean.user_id.toString(),false)
            }
//            itemView.user_id.text = "ID:" + bean.good_number.toString()
//            itemView.user_id.setCompoundDrawablesWithIntrinsicBounds(if (bean.good_number_state == 0) 0 else R.drawable.common_user_icon_liang, 0, 0 ,0)
            if (bean.auth != 0) {
                itemView.auth_tv.visibility = View.VISIBLE
                if (bean.auth == 2){
                    itemView.auth_tv.setImageResource(R.drawable.room_icon_user_flag_owner)
                }else{
                    itemView.auth_tv.setImageResource(R.drawable.room_icon_user_flag_manager)
                }
            } else {
                itemView.auth_tv.visibility = View.GONE
            }

            itemView.user_sex.setSeleted(bean.gender)
            itemView.mic_status.visibility = if (bean.status == 0) View.GONE else View.VISIBLE
            itemView.user_status.visibility = if (bean.speak == 0) View.GONE else View.VISIBLE
            itemView.mic_online.visibility = if (bean.type == 0) View.GONE else View.VISIBLE
            if (bean.mike < 9){
                itemView.mic_online.text = "${bean.mike}麦"
            }else{
                itemView.mic_online.text = "房主"
            }
            ImgUtil.loadFaceIcon(mContext, bean.face, itemView.user_icon)
            itemView.user_nick.text = bean.nickname
            itemView.setOnClickListener {
                mListener.onUserItemClick(it,bean)
            }
        }
    }
}

